src/app/public/modules/data-manager/data-manager.service.ts

```diff
@@ -179,13 +184,25 @@ export class SkyDataManagerService implements OnDestroy {
    * @param sourceId The ID of the entity subscribing to data state updates. This can be any value you choose
    * but should be unique within the data manager instance and should also be used when that entity updates the state.
    */
-  public getDataStateUpdates(sourceId: string): Observable<SkyDataManagerState> {
+  public getDataStateUpdates(
+    sourceId: string,
+    updateFilter?: SkyDataManagerStateUpdateFilterArgs
+    ): Observable<SkyDataManagerState> {
     // filter out events from the provided source and emit just the dataState
-    const dataStateObservable = this.dataStateChange.pipe(
-      filter(stateChange => sourceId !== stateChange.source),
-      map(stateChange => stateChange.dataState)
-    );
-    return dataStateObservable;
+    if (updateFilter) {
+      return this.dataStateChange.pipe(
+        filter(stateChange => sourceId !== stateChange.source),
+        map(stateChange => stateChange.dataState),
+        updateFilter.comparator ?
```

Comment:
Since the next two lines are part of the same statement, I'd indent them so they don't look like additional parameters to `pipe()`.

```
updateFilter.comparator ?
  distinctUntilChanged(updateFilter.comparator) :
  distinctUntilChanged(this.getDefaultStateComparator(updateFilter.properties))
```